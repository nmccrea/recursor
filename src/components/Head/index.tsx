import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import { Helmet } from "react-helmet"

interface Props {
  description?: string
  lang?: string
  meta?: Array<object>
  title: string
}

const Head = ({ description = ``, lang = `en`, meta = [], title }: Props) => {
  const { site } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
            description
            author
          }
        }
      }
    `
  )

  const metaDescription = description || site.siteMetadata.description

  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={title}
      titleTemplate={`%s | ${site.siteMetadata.title}`}
      meta={[
        {
          name: `description`,
          content: metaDescription,
        },
        {
          property: `og:title`,
          content: title,
        },
        {
          property: `og:description`,
          content: metaDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: site.siteMetadata.author,
        },
        {
          name: `twitter:title`,
          content: title,
        },
        {
          name: `twitter:description`,
          content: metaDescription,
        },
        ...meta,
      ]}
    />
  )
}

export default Head
