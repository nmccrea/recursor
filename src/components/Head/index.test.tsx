import { render } from "@testing-library/react"
import React from "react"
import { Helmet } from "react-helmet"
import Head from "."

describe("`<Head />`", () => {
  it("renders with the correct `react-helmet` state", () => {
    render(<Head title={"Test Title"} />)

    expect(Helmet.peek()).toMatchSnapshot()
  })
})
