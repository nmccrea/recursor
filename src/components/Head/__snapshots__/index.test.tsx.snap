// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<Head />\` renders with the correct \`react-helmet\` state 1`] = `
Object {
  "baseTag": Array [],
  "bodyAttributes": Object {},
  "defer": true,
  "encode": true,
  "htmlAttributes": Object {
    "lang": "en",
  },
  "linkTags": Array [
    Object {
      "href": "https://fonts.googleapis.com/css2?family=Audiowide&family=Roboto+Mono:wght@500&display=swap",
      "rel": "stylesheet",
    },
  ],
  "metaTags": Array [
    Object {
      "content": "A mock description.",
      "name": "description",
    },
    Object {
      "content": "Test Title",
      "property": "og:title",
    },
    Object {
      "content": "A mock description.",
      "property": "og:description",
    },
    Object {
      "content": "website",
      "property": "og:type",
    },
    Object {
      "content": "summary",
      "name": "twitter:card",
    },
    Object {
      "content": "Mockie McGee",
      "name": "twitter:creator",
    },
    Object {
      "content": "Test Title",
      "name": "twitter:title",
    },
    Object {
      "content": "A mock description.",
      "name": "twitter:description",
    },
  ],
  "noscriptTags": Array [],
  "onChangeClientState": [Function],
  "scriptTags": Array [],
  "styleTags": Array [],
  "title": "Test Title | Mock Site Name",
  "titleAttributes": Object {},
}
`;
